exchanges/kraken/kraken_test.go

```diff
@@ -1741,3 +1741,10 @@ func TestEnforceStandardChannelNames(t *testing.T) {
 		assert.ErrorIsf(t, err, subscription.ErrUseConstChannelName, "Private channel names should not be allowed for %s", n)
 	}
 }
+
+func TestWebsocketAuthToken(t *testing.T) {
+	t.Parallel()
+	k := new(Kraken)
+	k.wsAuthToken = "meep" // k is a local instance, so we can set this lock free
+	assert.Equal(t, "meep", k.websocketAuthToken())
```

Comment:
If what want to show here is how synchronized the test is by calling `websocketAuthToken()` method, it can be done with several similar calls running in parallel using `t.Run()`, I believe.

Reply:
🚧 I agree with the premise here; It's sane to ensure this exercises the lock.
However t.Run is definitely not the right way to achieve that, because there's zero guarantee of parallelism on the lock.
Something like:
```suggestion
k.wsAuthMutex.Lock()
c := make(chan string, 1)
go func(){
    c <- k.websocketAuthToken()
}()
runtime.GoSched()
select {
case <-time.After(time.Milisecond):
    // All good
case <-c:
    t.Error("websocketAuthToken must respect wsAuthMutex.Lock")
}
k.wsAuthMutex.Unlock()
```